### PR TITLE
Includes hdf5dev.h in hdf5.h

### PR DESCRIPTION
**Pull request by derobins**
Might want to ditch hdf5dev.h in the future and just directly include the H5Xdevelop.h files since hdf5dev.h is kind of a moot point with this change, but this PR at least allows the plugins to be built.

### CODE REVIEW

**Review by qkoziol — CHANGES_REQUESTED** [review]
I don't think this is the correct direction to go.   Why are you taking the hdf5dev.h header out of these files?   The whole point was to _not_ include this information in the header file for application developers (hdf5.h).

**Comment by derobins** [comment]
> I don't think this is the correct direction to go. Why are you taking the hdf5dev.h header out of these files? The whole point was to _not_ include this information in the header file for application developers (hdf5.h).

It's breaking all the daily tests that build external plugins. Semantically separating the information into separate header files is useful, but having every external thing navigate our new headers isn't a bridge we're going to cross at this time.

**Comment by qkoziol** [comment]
> > I don't think this is the correct direction to go. Why are you taking the hdf5dev.h header out of these files? The whole point was to _not_ include this information in the header file for application developers (hdf5.h).
> 
> It's breaking all the daily tests that build external plugins. Semantically separating the information into separate header files is useful, but having every external thing navigate our new headers isn't a bridge we're going to cross at this time.

I can look into this more, maybe another solution would suggest itself.   Could you send me an example or two of logs that show the failure(s)?

File: src/hdf5.h [hunk 1]
```diff
@@ -41,8 +41,20 @@
 #include "H5VLpublic.h" /* Virtual Object Layer                     */
 #include "H5Zpublic.h"  /* Data filters                             */
 
-/* "developer-oriented" headers (for VFD, VOL, etc. authors) */
-#include "hdf5dev.h"
+/* Application developer headers for various interfaces */
```
Comment (qkoziol) [inline src/hdf5.h]: These are for plugin authors - definitely not for application developers. :-)

**Comment by qkoziol** [comment]
Dana and I chatted about this a lot and this current iteration seems like a reasonable place to land.

**Review by qkoziol — APPROVED** [review]

File: src/hdf5.h (hunk 1 above)
Comment (derobins) [reply src/hdf5.h]: I copied that out of the hdf5dev.h header that you authored :)

Will change, though.

Comment (qkoziol) [reply src/hdf5.h]: Thanks - dunno where I was at with that, or where I pasted it from... :-)

**Review by byrnHDF — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

**Pull request merged**